src/NHibernate.Test/NHSpecificTest/NH3079/Fixture.cs

```diff
@@ -0,0 +1,235 @@
+				}
+
+				t.Commit();
+			}
+		}
+
+		// Test reproducing the problem.
+		[Test]
+		public void GetPersonTest()
+		{
+			using (var session = OpenSession())
+			{
+				var person1_2 = session.Get<Person>(new PersonCpId { IdA = 1, IdB = 2 });
+				Assert.That(person1_2.Name, Is.EqualTo("PERSON_1_2"));
+				Assert.That(
+					person1_2.EmploymentList.Select(e => e.Name),
+					Is.EquivalentTo(new[] { "EMPLOYMENT_13_1_1_7_8", "EMPLOYMENT_14_1_1_9_10" }));
+			}
+		}
+
+		// Test reproducing the problem.
+		[Test]
+		public void GetEmployerTest()
+		{
+			using (var session = OpenSession())
+			{
+				var employer7_8 = session.Get<Employer>(new EmployerCpId { IdA = 7, IdB = 8 });
+				Assert.That(employer7_8.Name, Is.EqualTo("EMPLOYER_7_8"));
+				Assert.That(
+					employer7_8.EmploymentList.Select(e => e.Name),
+					Is.EquivalentTo(new[] { "EMPLOYMENT_13_1_1_7_8" }));
+			}
+		}
+
+		[Test]
+		public void GetEmploymentTest()
+		{
+			using (var session = OpenSession())
+			{
+				var employment_13_1_2_7_8 =
+					session.Get<Employment>(
+						new EmploymentCpId
+						{
+							Id = 13,
+							PersonObj =
+								new Person
+								{
+									CpId = new PersonCpId { IdA = 1, IdB = 2 }
+								},
+							EmployerObj =
+								new Employer
+								{
+									CpId = new EmployerCpId { IdA = 7, IdB = 8 }
+								}
+						});
+				Assert.That(employment_13_1_2_7_8.Name, Is.EqualTo("EMPLOYMENT_13_1_1_7_8"));
+			}
+		}
+
+		[Test]
+		public void HqlPersonPositional()
+		{
+			using (var session = OpenSession())
+			{
+				var personList =
+					session
+						.GetNamedQuery("personPositional")
+						.SetParameter(0, new PersonCpId { IdA = 1, IdB = 2 })
+						.SetParameter(1, new PersonCpId { IdA = 3, IdB = 4 })
+						.List<Person>();
+				Assert.That(
+					personList.Select(e => e.Name),
+					Is.EquivalentTo(new[] { "PERSON_1_2", "PERSON_3_4" }));
+			}
+		}
+
+		[Test]
+		public void HqlPersonNamed()
+		{
+			using (var session = OpenSession())
+			{
+				var personList =
+					session
+						.GetNamedQuery("personNamed")
+						.SetParameter("id1", new PersonCpId { IdA = 1, IdB = 2 })
+						.SetParameter("id2", new PersonCpId { IdA = 3, IdB = 4 })
+						.List<Person>();
+				Assert.That(
+					personList.Select(e => e.Name),
+					Is.EquivalentTo(new[] { "PERSON_1_2", "PERSON_3_4" }));
+			}
+		}
+
+		[Test]
+		public void GetPersonNoComponent()
+		{
+			using (var session = OpenSession())
+			{
+				var person17_18 =
+					session.Get<PersonNoComponent>(new PersonNoComponent { IdA = 17, IdB = 18 });
+				Assert.That(person17_18.Name, Is.EqualTo("PERSON_NO_COMPONENT_17_18"));
+			}
+		}
+
+		[Test]
+		public void SqlPersonNoComponent()
+		{
+			using (var session = OpenSession())
+			{
+				var personList =
+					session
+						.GetNamedQuery("personNoComponentSql")
+						.SetParameter(0, new PersonNoComponent { IdA = 17, IdB = 18 })
+						.SetParameter(1, new PersonNoComponent { IdA = 19, IdB = 20 })
```

Comment:
The tests implying custom sql loaders only have one parameter. This test allows to check the change works also when multiple parameters needs flattening.